### ISSUE TITLE
PIDCache causes forever increasing memory usage

### ISSUE THREAD

**Issue by DenizPiri** [AsynkronIT/protoactor-dotnet#1301]
## Expected Behavior
When many objects that clean themselves are in use with Cluster, memory shouldn't forever grow.
PIDCache should self-maintain and clean itself.

## Actual Behavior
Memory usage forever grows. **PIDCache** grows forever. Each **PID** holds a **Process** reference, which causes **ActorProcess** and **RemoteProcess** instance count to grow.

## Steps to Reproduce the Problem

  1. Run the code in the following repo: https://github.com/DenizPiri/ProtoActorClusterPIDCacheMemoryLeak/blob/main/ProtoActorClusterMemoryLeak/Program.cs

## Specifications

  - Version: 0.25.0
  - Platform: Windows

We see this issue in our production instances. Using PidCacheInvalidation doesn't really help, as it is designed only to solve the problem of remote PIDs, local PIDs still stay in PIDCache forever, together with their ActorProcess instances.

**Comment by DenizPiri** [comment]
Currently, this pull request partially addresses the issue: https://github.com/asynkron/protoactor-dotnet/pull/1300

I believe, if the thoughts mentioned here would be implemented, it would be fully solved: https://github.com/asynkron/protoactor-dotnet/pull/1300#issuecomment-979163900

**Comment by DenizPiri** [comment]
https://github.com/asynkron/protoactor-dotnet/pull/1300#issuecomment-979901080 should fully resolve this issue right now. Once that is merged, this can be closed.

**Comment by DenizPiri** [comment]
#1300 is merged. Closing this one.
Once the new release is out, I will try it on production to verify that the RemoteProcess leak fix also works, if the issue persists, I will create a new issue or revive this one.
